Update README.md to reflect that the trust namespace is only for secrets (#1337)
cert-manager/website: content/docs/trust/trust-manager/README.md

One of the more important configuration options you might need to consider at install time is which "trust namespace" to use,
which can be set via the Helm value `app.trust.namespace`.

> The trust namespace is the only one in which `Secret` sources can be read. This restriction is in place

Reviewer: ```suggestion
By default, the trust namespace is the only one in which `Secret`s can be read. This restriction is in place
```

Reviewer: "By default" because trust-manager can be put into a mode where Secrets can be used as a target. https://github.com/cert-manager/trust-manager/blob/ad150e553b943a507949d183a3e71e869a2c5d83/deploy/charts/trust-manager/templates/clusterrole.yaml#L59

Author: Hmm, I'm not sure I like this wording as much, because it's not quite clear enough. I have another suggestion, will post momentarily.

> By default, the trust namespace is the only one in which `Secret`s can be read. This restriction is in place

Author: ```suggestion
By default, the trust namespace is the only namespace where`Secret`s will be read. This restriction is in place
```

> for security reasons - we don't want to give trust-manager the permission to read all `Secret`s in all namespaces.

Author: ```suggestion
for security reasons - we don't want to give trust-manager the permission to read all `Secret`s in all namespaces. With additional configuration, secrets may be read from or written to other namespaces.
```
